### PR TITLE
Disallow method invocation directly after xml navigation access expression.

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose
This is because according to the spec, method invocations after XML navigation expression (xmlVal/<elem>.method()) should be associated with the navigation expression.

This means when we add the method invocation support to XML navigation expressions, method invocation directly after the nav-expression will change the semantic. Hence this PR is disallowing it. If the user want to invoke a method on the result of an XML navigation expression they can use grouping expressions to group the navigation expression and invoke the method on the group.

Also related to https://github.com/ballerina-platform/ballerina-lang/issues/21567

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 1]
```diff
@@ -488,6 +488,10 @@ error.invalid.xml.ns.interpolation=\
 error.cannot.find.xml.namespace.prefix=\
   cannot find xml namespace prefix ''{0}''
 
+error.method.invocation.in.xml.navigation.expressions.not.supported=\
+  method invocations are not supported within XML navigation expressions yet, use grouping expression (parenthesis) \
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/resources/compiler.properties]:
```suggestion
  method invocations are not yet supported within XML navigation expressions, use a grouping expression (parenthesis) \
```

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 2]
```diff
@@ -488,6 +488,10 @@ error.invalid.xml.ns.interpolation=\
...
+  if you intent to invoke the method in the result of navigation expression.
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/resources/compiler.properties]:
```suggestion
  if you intend to invoke the method on the result of the navigation expression.
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java [hunk 3]
```diff
@@ -2102,6 +2102,15 @@ public void visit(BLangXMLNavigationAccess xmlNavigation) {
         if (xmlNavigation.childIndex != null) {
             analyzeExpr(xmlNavigation.childIndex);
         }
+        validateMethodInvocationsInXMLNavigationExpression(xmlNavigation);
+    }
+
+    private void validateMethodInvocationsInXMLNavigationExpression(BLangXMLNavigationAccess expression) {
+        if (!expression.methodInvocationAnalyzed && expression.parent.getKind() == NodeKind.INVOCATION) {
+            dlog.error(((BLangInvocation) expression.parent).name.pos,
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java]: Can't we use
```suggestion
            dlog.error(((BLangInvocation) expression.parent).pos,
```
instead?

Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java]: Wanted to get the error highlighting just for method name + paranthesis, but just realized that paranthesis will not be included in the position. Will changed to suggested.

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 4]
```diff
@@ -488,6 +488,10 @@ error.invalid.xml.ns.interpolation=\
 error.cannot.find.xml.namespace.prefix=\
   cannot find xml namespace prefix ''{0}''
 
+error.method.invocation.in.xml.navigation.expressions.not.supported=\
+  method invocations are not yet supported within XML navigation expressions yet, use a grouping expression \
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/resources/compiler.properties]:
```suggestion
  method invocations are not yet supported within XML navigation expressions, use a grouping expression \
```

**Pull request merged**